controllers/managedocs_controller.go

```diff
@@ -904,6 +975,15 @@ func (r *ManagedOCSReconciler) reconcilePrometheus() error {
 			Key: alertRelabelConfigSecretKey,
 		}
 
+		r.prometheus.Spec.Web.TLSConfig.ClientCA = promv1.SecretOrConfigMap{
+			ConfigMap: &corev1.ConfigMapKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: servingCertCABundleConfigMapName,
+				},
+				Key: "service-ca.crt",
+			},
+		}
```

Comment:
The main idea behind the templates is that the templates define the entire structure of the desired resource and include all of the data that is constant outside of any specific context. 

In accordance with that, this entire structure should go into the template (to be cloned when we clone the template using `DeepCopy`) and replaced with 
```
r.prometheus.Spec.Web.TLSConfig.ClientCA.ConfigMap.LocalObjectReference.Name = servingCertCABundleConfigMapName
```